### PR TITLE
Fix "Tuple domain handles must have assigned symbols" error

### PR DESCRIPTION
**Pull request by martint**
Fixes https://github.com/prestodb/presto/issues/9800

No test for now. It requires a connector that supports predicate pushdown and index joins.

### CODE REVIEW

**Comment by findepi** [comment]
@martint thanks for taking care of this!

> No test for now. It requires a connector that supports predicate pushdown and index joins.

TPCH supports predicate pushdown and Indexed-TPCH supports indexed joins. Should Indexed-TPCH gain support for pushdown?
cc @kokosing @sopel39

**Comment by sopel39** [comment]
+1 if it would allow to write tests

**Comment by kokosing** [comment]
Indexed TPCH supports predicate pushdown as it extends normal TPCH (`TpchIndexMetadata extends TpchMetadata`). I could try to add such test.

**Comment by kokosing** [comment]
> Indexed TPCH supports predicate pushdown as it extends normal TPCH (TpchIndexMetadata extends TpchMetadata). I could try to add such test.

Though it needs to be enabled ;)

**Comment by findepi** [comment]
@kokosing maybe we can convert `TestDistributedQueriesIndexed` for that purpose?

**Comment by kokosing** [comment]
See https://github.com/prestodb/presto/pull/9823, if you apply this PR and revert `3eac47dfec` then you will get an error mentioned in #9800

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/IndexSourceNode.java [hunk 1]
```diff
@@ -67,11 +67,6 @@ public IndexSourceNode(
         checkArgument(!outputSymbols.isEmpty(), "outputSymbols is empty");
         checkArgument(assignments.keySet().containsAll(lookupSymbols), "Assignments do not include all lookup symbols");
         checkArgument(outputSymbols.containsAll(lookupSymbols), "Lookup symbols need to be part of the output symbols");
-        Set<ColumnHandle> assignedColumnHandles = ImmutableSet.copyOf(assignments.values());
-        currentConstraint.getDomains().ifPresent(handleToDomain ->
-                checkArgument(
-                        assignedColumnHandles.containsAll(handleToDomain.keySet()),
-                        "Tuple domain handles must have assigned symbols"));
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/IndexSourceNode.java]: I was not confident enough to remove this.

**Pull request merged**